howso/direct/client.py

```diff
@@ -1099,19 +1072,20 @@ def release_trainee_resources(self, trainee_id: str):
         try:
             cache_item = self.trainee_cache.get_item(trainee_id)
             trainee = cache_item['trainee']
-            if cache_item.get('entity_id', self.howso.handle) == self.howso.handle:
-                if trainee.persistence in ['allow', 'always']:
-                    # Persist on unload
-                    self.howso.persist(trainee_id)
-                elif trainee.persistence == "never":
-                    raise HowsoError(
-                        "Trainees set to never persist may not have their "
-                        "resources released. Delete the Trainee instead.")
-                self.trainee_cache.discard(trainee_id)
+
+            if trainee.persistence in ['allow', 'always']:
+                # Persist on unload
+                self.howso.persist(trainee_id)
+            elif trainee.persistence == "never":
+                raise HowsoError(
+                    "Trainees set to never persist may not have their "
+                    "resources released. Delete the Trainee instead.")
         except KeyError:
             # Trainee not cached, ignore
             pass
-        self.howso.delete(trainee_id)
+        finally:
+            self.trainee_cache.discard(trainee_id)
+            self.howso.delete(trainee_id)
```

Comment:
I am a bit concerned with doing these both down here in a finally. Since if we fail to persist you now have lost changes, assuming your script catches errors thrown or you are in a notebook. Additionally, we don't really need to discard from the cache always either since if persistence is never its intended for you to delete it. But the cache discarding shouldn't negatively hurt anything since it would just be re-loaded into the cache if needed.